### PR TITLE
Some other small fixes for the profile page.

### PR DESCRIPTION
**Pull request by frankban**
Changes allow showing bundles in the profile page.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5475/
Test PASSed.

**Comment by anthonydillon** [comment]
Code looks good :+1:. We should have a follow up card for the error notification work.

File: jujugui/static/gui/src/app/components/user-profile/entity/entity.js [hunk 1]
```diff
@@ -231,8 +231,8 @@ YUI.add('user-profile-entity', function() {
       }
       var entity = this.props.entity;
       var services = [];
-      var names = Object.keys(entity.services);
-      names.forEach((name) => {
+      var applications = entity.applications || entity.services || {};
+      Object.keys(applications).forEach((name) => {
```
Comment (urosj) [inline jujugui/static/gui/src/app/components/user-profile/entity/entity.js]: Does this work with Juju 1.2x?

File: jujugui/static/gui/src/app/components/user-profile/user-profile.js [hunk 2]
```diff
@@ -152,8 +152,12 @@ YUI.add('user-profile', function() {
     _listBudgetsCallback: function(error, data) {
       this.setState({loadingBudgets: false});
       if (error) {
-        // TODO huwshimi: notify the user with the error.
-        console.error('cannot retrieve terms:', error);
+        if (error.indexOf('not found') === -1) {
```
Comment (urosj) [inline jujugui/static/gui/src/app/components/user-profile/user-profile.js]: TY

File: jujugui/static/gui/src/app/components/user-profile/user-profile.js [hunk 3]
```diff
@@ -555,9 +559,10 @@ YUI.add('user-profile', function() {
     _generateBundleRow: function(bundle) {
       var id = bundle.id;
       var services = [];
-      var serviceNames = Object.keys(bundle.services);
+      var applications = bundle.applications || bundle.services || {};
+      var serviceNames = Object.keys(applications);
       serviceNames.forEach((serviceName, idx) => {
-        var service = bundle.services[serviceName];
+        var service = applications[serviceName];
```
Comment (urosj) [inline jujugui/static/gui/src/app/components/user-profile/user-profile.js]: Same consideration as above for juju 1.2x

**Comment by huwshimi** [comment]
Looks good, might be worth adding a couple of tests for the different keys to make sure this works and doesn't regress.

File: jujugui/static/gui/src/app/components/user-profile/user-profile.js (hunk 3 above)
Comment (frankban) [inline jujugui/static/gui/src/app/components/user-profile/user-profile.js]: I cannot find the consideration above. but the code falls back to .services so Juju 1 should be ok.

**Comment by fabricematrat** [comment]
:+1: on sandbox and gijoe 2.0 beta 12

**Comment by frankban** [comment]
Thnak you all for the reviews!
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**